src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -432,6 +432,58 @@ int SystemProcessInterface::SystemProcesses::system_processes(SystemProcess** sy
   *no_of_sys_processes = process_count;
   *system_processes = next;
 
+  return OS_OK;
+#elif defined(__FreeBSD__)
+  struct kinfo_proc *lproc;
+  int mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_ALL };
+  const size_t miblen = 3;
+  size_t length;
+  int pid_count;
+
+  if (sysctl(mib, miblen, NULL, &length, NULL, 0) == -1) {
+    return OS_ERR;
+  }
+
+  lproc = (struct kinfo_proc *)malloc(length);
+  if (!lproc) {
+    return OS_ERR;
+  }
+
+  if (sysctl(mib, miblen, lproc, &length, NULL, 0) == -1) {
+    free(lproc);
+    return OS_ERR;
+  }
+
+  pid_count = length / sizeof(*lproc);
+  int process_count = 0;
+  SystemProcess *next;
+  
+  for (int i = 0; i < pid_count; i++) {
+     int bmib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, lproc[i].ki_pid };
```

Comment:
I'm curious as to what led you to choose "bmib" as the variable name here?  I was thinking "pmib" maybe would be clearer, but I'd like to know the reasoning first :)

Reply:
b as second request but I can change to p no worries 😉